libs/src/sdk/sdk.ts

```diff
@@ -153,18 +153,22 @@ const initializeApis = ({
 }) => {
   const initializationPromise = discoveryProvider.init()
 
-  const fetchApi = async (url: string) => {
+  const fetchApi = async (url: string, context?: RequestInit) => {
     // Ensure discovery node is initialized
     await initializationPromise
 
     // Append the appName to the query params
     const urlWithAppName =
       url + (url.includes('?') ? '&' : '?') + querystring({ app_name: appName })
 
+    const requestParams: Record<string, unknown> = {}
+    if (!!context) {
+      Object.assign(requestParams, context)
+    }
+    requestParams['endpoint'] = urlWithAppName
```

Comment:
I'm guessing the linter will want `requestParams.endpoint`

Reply:
Don't need this anymore but I actually get a complaint if I use `requestParams.endpoint`.
`Property 'endpoint' comes from an index signature, so it must be accessed with ['endpoint'].`
I think because I typed `requestParams` as a `Record`.